plugins/woocommerce-admin/client/products/product-form-actions.tsx

```diff
@@ -35,6 +36,8 @@ export const ProductFormActions: React.FC = () => {
 	const { isDirty, isValidForm, values, resetForm } =
 		useFormContext< Product >();
 
+	usePreventLeavingPage( isDirty );
```

Comment:
`Changes you made may not be saved.` is this the default text that shows up in the popup? I was confused where this was coming from.
Also I assume `Changes you made may not be saved.` will be supported by different languages, as the browser will take care of that? Otherwise I did say we should pass ours in manually.